plugins/quota/rest.go

```diff
@@ -2,29 +2,103 @@ package quota
 
 import (
 	"net/http"
+	"os"
+	"path"
 
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
 	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/pkg/api"
 	"github.com/gluster/glusterd2/pkg/errors"
 	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
 )
 
-func commonValidator(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	volname := mux.Vars(r)["volname"]
-	v, err := volume.GetVolume(volname)
+//QuotadStart is the function to start the quota daemon
+func QuotadStart(c transaction.TxnCtx) error {
+	quotadDaemon, err := NewQuotad()
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
-	} else if v.State != volume.VolStarted {
-		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return err
+	}
+	// Create pidfile dir if not exists
+	err = os.MkdirAll(path.Dir(quotadDaemon.pidfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
+	}
+	// Create logFiledir dir
+	err = os.MkdirAll(path.Dir(quotadDaemon.logfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
 	}
+	err = daemon.Start(quotadDaemon, true)
+	return nil
 }
 
 func quotaEnableHandler(w http.ResponseWriter, r *http.Request) {
-	// implement the help logic and send response back as below
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volName := p["volname"]
+
 	ctx := r.Context()
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "todo: quota enable")
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate volume existence
+	vol, err := volume.GetVolume(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	if err := txn.Ctx.Set("volinfo", vol); err != nil {
```

Comment:
No change to the volinfo! before `txn.Ctx.Set` update volinfo with Quota enabled flag or option. You can use that info when enable called second time on the same volume. 

    if vol.quotaEnabled { return already_enabled_error}

You can also introduce `force` option, with force you can allow to try starting Quota again.

Reply:
done. will introduce force in the oncoming patches